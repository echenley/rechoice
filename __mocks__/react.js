const React = require.requireActual('react')

React.Fragment = 'Fragment'

module.exports = React
