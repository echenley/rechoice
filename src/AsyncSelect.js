// @flow

import * as React from 'react'
import Async from './Async'
import Select from './Select'
import type {
  FilterItemsSingle,
  LoadItems,
  Item,
  LoadingState,
  ItemRenderer,
  MenuRenderer,
} from './types'

type Props = {
  disabled?: boolean,
  filterItems?: FilterItemsSingle,
  getInputValue: (item: ?Item) => string,
  itemKey: string,
  loadingIndicator?: React.Node,
  loadingState?: LoadingState,
  loadItems: LoadItems,
  minimumInput?: number,
  onBlur?: () => mixed,
  onChange: (item: ?Item) => mixed,
  onFocus?: () => mixed,
  onInputChange?: (inputValue: string) => mixed,
  placeholder?: string,
  renderItem: ItemRenderer,
  renderMenu?: MenuRenderer,
  value: ?Item,
}

const AsyncSelect = ({
  loadItems,
  onChange,
  onFocus,
  onInputChange,
  ...props
}: Props) => {
  return (
    <Async
      loadItems={loadItems}
      minimumInput={props.minimumInput}
      onChange={onChange}
      onFocus={onFocus}
      onInputChange={onInputChange}
    >
      {asyncProps => <Select {...props} {...asyncProps} />}
    </Async>
  )
}

export default AsyncSelect
