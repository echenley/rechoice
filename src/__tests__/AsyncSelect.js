import * as React from 'react'
import { shallow } from 'enzyme'
import AsyncSelect from '../AsyncSelect'

const renderComponent = props => {
  return shallow(
    <AsyncSelect
      itemKey="id"
      loadItems={() => Promise.resolve({ items: [] })}
      loadingIndicator={<div className="loading-indicator" />}
      onChange={() => {}}
      onFocus={() => {}}
      placeholder="Placeholder"
      renderItem={() => <div />}
      renderSelected={item => item.id}
      value={null}
      {...props}
    />
  )
}

describe('AsyncSelect', () => {
  it('renders correctly', () => {
    expect(renderComponent().getElement()).toMatchSnapshot()
  })

  it('renders a Select child', () => {
    expect(
      renderComponent()
        .dive()
        .getElement()
    ).toMatchSnapshot()
  })
})
