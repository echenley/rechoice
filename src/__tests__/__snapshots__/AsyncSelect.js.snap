// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsyncSelect renders a Select child 1`] = `
<Select
  autoResizeInput={false}
  disabled={false}
  filterItems={[Function]}
  itemKey="id"
  items={Array []}
  loadingIndicator={
    <div
      className="loading-indicator"
    />
  }
  loadingState="success"
  minimumInput={0}
  onChange={[Function]}
  onFocus={[Function]}
  onInputChange={[Function]}
  placeholder="Placeholder"
  renderItem={[Function]}
  renderSelected={[Function]}
  resetInputOnBlur={true}
  value={null}
/>
`;

exports[`AsyncSelect renders correctly 1`] = `
<Async
  loadItems={[Function]}
  minimumInput={0}
  onChange={[Function]}
  onFocus={[Function]}
  onInputChange={undefined}
>
  [Function]
</Async>
`;
